baseplate/lib/thrift_pool.py

```diff
@@ -203,10 +204,16 @@ def _is_stale(self, prot: TProtocolBase) -> bool:
         return False
 
     def _release(self, prot: Optional[TProtocolBase]) -> None:
-        if prot and prot.trans.isOpen():
```

Comment:
I'm assuming that `isOpen` can context switch, still trying to verify that.

Reply:
Looks like it actually peeks at message bytes on the socket to check if open, could very well context switch.

https://github.com/apache/thrift/blob/master/lib/py/src/transport/TSocket.py#L76